src/frontend/src/instance.rs

```diff
@@ -132,23 +131,29 @@ impl Instance {
         let partition_manager = Arc::new(PartitionRuleManager::new(table_routes));
         let datanode_clients = Arc::new(DatanodeClients::default());
 
-        let catalog_manager = Arc::new(FrontendCatalogManager::new(
-            meta_backend,
-            partition_manager,
-            datanode_clients.clone(),
-        ));
+        let mut catalog_manager =
+            FrontendCatalogManager::new(meta_backend, partition_manager, datanode_clients.clone());
 
-        let dist_instance =
-            DistInstance::new(meta_client, catalog_manager.clone(), datanode_clients);
+        let dist_instance = DistInstance::new(
+            meta_client,
+            Arc::new(catalog_manager.clone()),
```

Comment:
Can we refactor this to avoid circular reference?

Reply:
I've left a todo in the `dist_instance` field of FrontendCatalogManager. Once I've finished the work of creating distributed table by procedure, the circular dependency here will be eliminated.